src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/HealthCertificateValidationService.swift

```diff
@@ -62,15 +62,33 @@ final class HealthCertificateValidationService: HealthCertificateValidationProvi
 	) {
 		
 		// 1. Apply technical validation
-		let expirationDate = healthCertificate.cborWebTokenHeader.expirationTime
 		
+		let isExpired: Bool
+		let stub = DCCSignatureVerifiableStub(error: nil)
+		// DONT FORGET: pass singatures array into this function!
+		let result = stub.verify(certificate: healthCertificate.base45, with: [], and: validationClock)
+		switch result {
+		case .success:
+			isExpired = false
+		case.failure:
+			isExpired = true
+		}
+		
+		let expirationDate = healthCertificate.cborWebTokenHeader.expirationTime
 		// NOTE: We expect here a HealthCertificate, which was already json schema validated at its creation time. So the JsonSchemaCheck will here always be true. So we only have to check for the expirationTime of the certificate.
-		guard expirationDate >= validationClock else {
-			Log.warning("Technical validation failed: expirationDate < validationClock. Expiration date: \(private: expirationDate), validationClock: \(private: validationClock)", log: .vaccination)
-			completion(.failure(.TECHNICAL_VALIDATION_FAILED))
+		let signatureInvalid = expirationDate < validationClock
+		
+		guard !isExpired && !signatureInvalid else {
+			if isExpired {
+				Log.warning("Technical validation failed: signature invalid.", log: .vaccination)
```

Comment:
According to the [tech spec](https://github.com/corona-warn-app/cwa-app-tech-spec/pull/80/files#diff-88aaaad167d935e204c7aea038c57b61081a1ad11163dee7b61aa4f85cc5f469) we should log the error returned from the signature verification service here.

Reply:
Done